### PR TITLE
Added dockerconfig workspace and corrected regex expression for all 3 tasks.

### PR DESCRIPTION
**Pull request by Senjuti256**
Added the optional dockerconfig workspace to the S2I task to provide flexibility to the customers to provide their dockerfile and also to make our task at par with the S2I clustertasks.  
Signed off by:- @Senjuti256 <sde@redhat.com>

### CODE REVIEW

File: values.yaml [hunk 1]
```diff
@@ -4,8 +4,8 @@ images:
   bash: registry.access.redhat.com/ubi8-minimal:8.9
   skopeo: registry.access.redhat.com/ubi8/skopeo:8.9
   buildah: registry.access.redhat.com/ubi8/buildah:8.9
-  s2i: registry.access.redhat.com/source-to-image/source-to-image-rhel8:v1.3.9-6
-
+  s2i: registry.redhat.io/source-to-image/source-to-image-rhel8:v1.3.9-6
```
Comment (vdemeester) [inline values.yaml:7]: Both work, don't they ?

Comment (Senjuti256) [reply values.yaml:7]: Nope the `registry.access.redhat.com` doesn't work @vdemeester .

Comment (vdemeester) [reply values.yaml:7]: On CI it works / worked, otherwise we wouldn't had a green CI 😝

Comment (savitaashture) [reply values.yaml:7]: I agree with @vdemeester both will work

Comment (Senjuti256) [reply values.yaml]: @vdemeester @savitaashture made the required changes.

**Pull request merged**